src/diffusers/models/modeling_utils.py

```diff
@@ -562,6 +623,41 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
             **kwargs,
         )
 
+        # Determine if we're loading from a directory of sharded checkpoints.
+        is_sharded = False
+        is_local = os.path.isdir(pretrained_model_name_or_path)
+        if is_local:
+            index_file = Path(
+                pretrained_model_name_or_path,
+                subfolder,
+                _add_variant(SAFE_WEIGHTS_INDEX_NAME if use_safetensors else WEIGHTS_INDEX_NAME, variant),
+            ).as_posix()
+            if os.path.isfile(index_file):
+                is_sharded = True
```

Comment:
You can also use `Path.is_file` here

```suggestion
            )
            if index_file.is_file():
                is_sharded = True
```